modules/settings.js

```diff
@@ -13,63 +12,120 @@ function Settings() {
+Settings.prototype.validate = function(potentialOptions, value) {
 
-    if (myDictionary.photos !== undefined &&
-      typeof myDictionary.photos === 'boolean') {
-      options.photos = myDictionary.photos;
-    }
+  var valid = true;
+  var opts = null;
+  var validTypes = {
+    photos: 'boolean',
+    videos: 'boolean',
+    contacts: 'boolean',
+    text: 'boolean',
+    intervalTime: 'number',
+    id: 'number',
+    registeredTimer: 'boolean',
+    repeat: 'boolean'
+  };
 
-    if (myDictionary.videos !== undefined &&
-      typeof myDictionary.videos === 'boolean') {
-      options.videos = myDictionary.videos;
+  if (typeof potentialOptions === 'object') {
+    opts = potentialOptions; // validate parameter object
+  } else if (typeof potentialOptions === 'string') {
+    opts = {}; // single field check
+    if (typeof value === 'undefined') {
+      opts[potentialOptions] = this.options[potentialOptions];
+    } else {
+      opts[potentialOptions] = value;
     }
+  } else if (typeof potentialOptions === 'undefined') {
+    opts = this.options; // validate current options
+  } else {
+    // TODO - replace with ErrorHandler module
+    return console.log('Invalid validate parameter', field);
+  }
 
-    if (myDictionary.contacts !== undefined &&
-      typeof myDictionary.contacts === 'boolean') {
-      options.contacts = myDictionary.contacts;
-    }
+  // Support partial validation
+  for (var field in opts) {
+    if (typeof this.options[field] === 'undefined') {
+      // TODO - replace with ErrorHandler module
+      console.log('Unrecognized settings option', field);
+      valid = false;
+    } else if (typeof opts[field] !== validTypes[field]) {
+      // TODO - replace with ErrorHandler module
+      console.log('Invalid type for settings option', field);
 
-    if (myDictionary.text !== undefined &&
-      typeof myDictionary.text === 'boolean') {
-      options.text = myDictionary.text;
-    }
 
-    if (myDictionary.id !== undefined &&
-      typeof myDictionary.id === 'number') {
-      options.id = myDictionary.id;
-    }
+      // alert(field + ': ' + typeof opts[field] + ' vs ' + typeof validTypes[field]); //rmv
 
-    if (myDictionary.registeredTimer !== undefined &&
-      typeof myDictionary.registeredTimer === 'boolean') {
-      options.registeredTimer = myDictionary.registeredTimer;
-    }
 
-    if (myDictionary.repeat !== undefined &&
-      typeof myDictionary.repeat === 'boolean') {
-      options.repeat = myDictionary.repeat;
-    }
-    //////pass in the value in hours /////////
-    if (myDictionary.intervalTime !== undefined &&
-      typeof myDictionary.intervalTime === 'number') {
-      timeInMilliSec = myDictionary.intervalTime * 1000 * 60 * 60;
-      options.intervalTime = timeInMilliSec;
+      valid = false;
     }
+  }
 
-    localStorage.setItem('ffosbrOptions', JSON.stringify(options));
-    return options;
-  };
-}
+  return valid;
+};
+
+Settings.prototype.set = function(newOptions) {
```

Comment:
My impression was that that was how you'd update settings through the UI. If I was wiring up a button to turn on photo backup, I was thinking the event handler would just call something like set('photos', true).

If we had another plan, though, that's fine – since we have to write the whole settings object to disk anyway, that layer of abstraction isn't strictly necessary.